### PR TITLE
Add E2E tests

### LINKED ISSUE #12: Add end-to-end tests
[Cypress](https://www.cypress.io/) would be a good option for that.

### PR DESCRIPTION
**Pull request by jmpressman**
Closes: #12 

Should we add CI integration as well or save for another PR?

### CODE REVIEW

**Review by t8p — COMMENTED** [review]
Looks good to me.

Btw: Is there a way with Cypress to test the features when SmartSelect is disabled? Because if we want to integrate CircleCI, there are always some tests missing.

**Comment by jmpressman** [comment]
> Btw: Is there a way with Cypress to test the features when SmartSelect is disabled? Because if we want to integrate CircleCI, there are always some tests missing.

Yes, initially I wanted to do that, but that would require the example apps to use the `viewCustom` method, which kind of runs counter to the idea that we keep our example apps lightweight. As mentioned  earlier, I am thinking of doing some of the viewConfig stuff differently when we add the ability to style via elm (vs css), so I held off on testing disability until then.

File: cypress/integration/multi_select_spec.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      cy.visit('./examples/MultiSelect/index.html')
+    })
+  
+    // describe('disabled', () => {
+    //   it('cannot be opened by clicking on it', () => {
+    //     // disable select first
+    //   })
+    // })
+  
+    describe('enabled', () => {
```
Comment (mschindlerMM) [inline cypress/integration/multi_select_spec.js]: Remove unnecessary code comments and provide a better description for the 'describe'

File: cypress/integration/multi_select_spec.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  
+        cy.get('#elm-smart-select--select-options-container')
+          .children()
+          .first()
+          .click()
+          .click()
+
+        cy.get('.elm-smart-select--multi-selected-container')
+          .children()
+          .first()
```
Comment (mschindlerMM) [inline cypress/integration/multi_select_spec.js]: These three lines seem to appear quiet frequently, so you could extract them into a helper or command function to improve readability

File: cypress/integration/single_select_spec.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
+context('Single Select', () => {
+  beforeEach(() => {
+    cy.visit('./examples/SingleSelect/index.html')
+  })
+
+  // describe('disabled', () => {
+  //   it('cannot be opened by clicking on it', () => {
+  //     // disable select first
+  //   })
+  // })
```
Comment (mschindlerMM) [inline cypress/integration/single_select_spec.js]: Remove code comments

**Review by mschindlerMM — COMMENTED** [review]
Just small notes on things I would like to see addressed - otherwise approved

**Comment by mschindlerMM** [comment]
> Should we add CI integration as well or save for another PR?

It probably makes sense to have a small CI setup since we now have specs to run. But I don't see that urgent and don't think that should be tackled in this branch. Also I wouldn't go for CircleCI for this project and just use Github Actions. Should be easy to setup, less infrastructure friction and we wouldn't use our CircleCI credits in openSource projects

**Pull request merged**